pallets/eth-bridge/src/tests/tests.rs

```diff
@@ -90,3 +94,164 @@ fn run_checks(
         assert_eq!(calldata, expected_calldata);
     })
 }
+
+#[cfg(test)]
+mod add_confirmation {
+
+    use super::*;
+    use frame_support::assert_ok;
+    use frame_system::RawOrigin;
+
+    #[test]
+    fn it_adds_confirmation_correctly() {
+        let mut ext = ExtBuilder::build_default().with_validators().as_externality();
+        ext.execute_with(|| {
+            let context = setup_context();
+
+            let tx_id = add_new_request::<TestRuntime>(
+                &context.request_function_name,
+                &context.request_params,
+            )
+            .unwrap();
+
+            let active_tx = ActiveTransaction::<TestRuntime>::get()
+                .expect("Active transaction should be present");
+
+            assert_ok!(EthBridge::add_confirmation(
+                RawOrigin::None.into(),
+                tx_id,
+                context.confirmation_signature.clone(),
+                context.confirming_author.clone(),
+                context.test_signature
+            ));
+
+            let active_tx = ActiveTransaction::<TestRuntime>::get()
+                .expect("Active transaction should be present");
+
+            assert!(
+                active_tx.confirmations.contains(&context.confirmation_signature),
+                "Confirmation should be present"
+            );
+
+            assert_eq!(
+                active_tx.data.sender, context.author.account_id,
+                "Sender should be the author's account_id"
+            );
+        });
+    }
+}
+
+#[cfg(test)]
+mod add_eth_tx_hash {
+    use super::*;
+    use frame_support::assert_ok;
+    use frame_system::RawOrigin;
+
+    fn run_eth_tx_hash_test(setup_fn: Option<fn(&mut ActiveTransactionData<TestRuntime>)>, expected: DispatchResultWithPostInfo) {
+        let mut ext = ExtBuilder::build_default().with_validators().as_externality();
+        ext.execute_with(|| {
+            let context = setup_context();
+
+            let tx_id = add_new_request::<TestRuntime>(
+                &context.request_function_name,
+                &context.request_params,
+            )
+            .unwrap();
+
+            if let Some(setup_fn) = setup_fn {
+                let mut active_tx = ActiveTransaction::<TestRuntime>::get().expect("is active");
+                setup_fn(&mut active_tx);
+                ActiveTransaction::<TestRuntime>::put(active_tx);
+            }
+
+            let result = EthBridge::add_eth_tx_hash(
+                RawOrigin::None.into(),
+                tx_id,
+                context.eth_tx_hash,
+                context.author,
+                context.test_signature.clone()
+            );
+
+            assert_eq!(result, expected);
+        });
+    }
+
+    #[test]
+    fn it_returns_error_when_eth_tx_hash_already_set() {
```

Comment:
is this related to `add_eth_tx_hash`, which is the name of the module?
 
Could be named `eth_tx_hash_cannot_be_set_twice`

Reply:
I think the test would become clearer if the modifying of parameters, which is done in the function passed to `run_eth_tx_hash_set` were somehow done on the main body of the test. That could be a field in `Context`, for example, and a well named function would modify that value.